### ISSUE TITLE
building examples, python, and vamp

### ISSUE THREAD

**Issue by sildeag** [MTG/essentia#726]
In `src/wscript` it appears the examples are not being built.

`if ctx.env.EXAMPLE_LIST or ctx.env.WITH_VAMP:`
` `        `ctx.recurse('examples')`

`if ctx.env.WITH_PYTHON:`
` `        `ctx.recurse('python')`

Is at the end of the file and not reached.

**Comment by dbogdanov** [comment]
I can't reproduce this issue, can you provide more details? Building examples, python bindings and vamp works just fine with the current master (using the --with-examples --with-python flags in ```waf configure``` to reach those lines of code).

**Comment by sildeag** [comment]
I tried it with a clone of MTG dated Friday last week and it works just fine now.  From my point-of-view tabs may have shifted the code to be unreachable, which was the issue.

The other issue, was related to linking inside a docker container, since the /usr/local/lib did not have an existing copy of essentia, I used '..' to get the copy from the folder one level up in the current build - this affected all examples.  I changed my build scripts to make it more explicit whether I was using `stlib`, `stlibpath`, `lib`, or `libpath` since I read up on `waf` since `static builds` seemed important to MTG.
